test/save_network_test.rb

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+require "y2storage"
+
+require "yast"
+require "network/clients/save_network"
+describe Yast::SaveNetworkClient do
+  describe "#adjust_for_network_disks" do
+    before do
+      allow(Yast::SCR).to receive(:Execute).and_return("exit" => 0, "stdout" => "", "stderr" => "")
+      fake_scenario(scenario)
+    end
+    let(:scenario) { "mixed_disks" }
```

Comment:
You are mocking the staging devicegraph and the only relevant call to it below. Thus, you don't need a YAML file describing the scenario, you don't need `fake_scenario` and, of course, you don't need all the infrastructure you introduced around it. This should be enough:

```ruby
describe Yast::SaveNetworkClient do
  describe "#adjust_for_network_disks" do
    before do
      Y2Storage::StorageManager.create_test_instance

      allow(Yast::SCR).to receive(:Execute).and_return("exit" => 0, "stdout" => "", "stderr" => "") 
      staging = Y2Storage::StorageManager.instance.y2storage_staging
      allow(staging).to receive(:filesystem_in_network?).and_return(in_network)
    end 

    context "when installation directory is in a network device" do
      let(:in_network) { true }

      it "tunes ifcfg file for remote filesystem" do
        expect(Yast::SCR).to receive(:Execute).with(anything, /nfsroot/).once
        subject.main
      end 
    end 

    context "when installation directory is in a local device" do
      let(:in_network) { false }

      it "does not touch any configuration file" do
        expect(Yast::SCR).to_not receive(:Execute).with(anything, /nfsroot/)
        subject.main
      end 
    end 
  end 
end
```

The rest of your test will keep working as-is. And you can revert all the changes done to the other tests and kill the YAML file and `storage_helpers.rb`.